### ISSUE TITLE
For `getDataObject()`: allow argument of `limit=-1`

### ISSUE THREAD

**Issue by gothub** [DataONEorg/rdataone#284]
Specifying `limit=-1` will have the special meaning to always download an item, regardless of it's size.

This is preferred over specifying a large value, for example `limit=1TB`.
This feature will be used for resource maps, which should always be downloaded by `getDataPackage()`.

If there are use cases for which downloading resources maps isn't desired, then I suggest that we develop
a different tool for those use cases.

**Comment by jeanetteclark** [comment]
yeah I can't think of a reason to not grab the resource map so that seems reasonable Peter

**Comment by gothub** [comment]
Do you get many large metadata documents? I'm assuming that the 'limit=' argument is intended for data objects, so may not always be relevant for metadata. Also, is it always necessary to download the metadata object for your use cases? It's probably needed when you want to update it, yes?

Would it be useful to discuss this at the ADC meeting tomorrow for a few minutes?
